### PR TITLE
feat(dgeni,daffio): render package API doc exports

### LINKED ISSUE #2816: [BUG] Daffodil Package API page missing child symbols
https://next.daff.io/docs/api/content

Generate correct data with dgeni

### LINKED ISSUE #2971: [FEAT] Set up main API page for each package
# :bulb: Feature request

## Feature Name
Set up main API page for each package

## The Desired Behavior
Main API page for each package should include a brief description of what the package is along with a comprehensive list of all the available APIs in the package.

## Your Use Case
As a user, I want to see an overview of all the available APIs in a package.

## Environment

<pre><code>
Daffodil version: X.Y.Z
Angular version: X.Y.Z 

 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: #2816, fixes: #2971

## What is the new behavior?
- generates package docs as JSON
- handles API package docs specially in the doc viewer
- extracts the API list children to a separate component for use with API package docs

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: apps/daffio/src/app/docs/api/components/api-list-children/api-list-children.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+@for (childDoc of children; track childDoc.id) {
```
Comment (xelaint) [inline apps/daffio/src/app/docs/api/components/api-list-children/api-list-children.component.html]: can we name this `api-list-section` instead?

File: apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.component.html [hunk 2]
```diff
@@ -3,10 +3,18 @@
...
 </button>
 <div class="daffio-doc-viewer__grid">
-	<daff-article
-		class="daffio-doc-viewer__content"
-		[innerHtml]="sanitizedContent">
-	</daff-article>
+  @if (isApiPackage) {
+    <daff-article class="daffio-doc-viewer__content">
+      <h1>{{doc.title}}</h1>
+      <h2>Exports</h2>
```
Comment (xelaint) [inline apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.component.html]: Why do we need to label it as exports?

**Comment by xelaint** [comment]
@griest024 I also need to make updates to the UI of list items in the api list. LMK when I can do that.

File: apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.component.html (hunk 2 above)
Comment (griest024) [reply apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.component.html]: The page looked really empty without it honestly but you can do whatever design seems best to you; I don't have any strong preference.

**Comment by griest024** [comment]
@xelaint you can go ahead and make UI updates

File: apps/daffio/src/app/docs/api/components/api-list-children/api-list-children.component.html (hunk 1 above)
Comment (griest024) [reply apps/daffio/src/app/docs/api/components/api-list-children/api-list-children.component.html]: sure, feel free to do this during your ui updates

File: apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.component.html (hunk 2 above)
Comment (xelaint) [reply apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.component.html]: Can we add a sidebar that lists out all the packages, similar to the individual `docs/packages` pages (w/ links to the package api page)

Comment (griest024) [reply apps/daffio/src/app/docs/components/doc-viewer/doc-viewer.component.html]: yeah I think that should be a different PR, can you add an issue for it?

**Comment by xelaint** [comment]
Screen Shot 2024-08-21 at 3 05 53 PM

@griest024 Is there a way to pull this section of the readme and add it to the page?

**Comment by griest024** [comment]
> Screen Shot 2024-08-21 at 3 05 53 PM
> @griest024 Is there a way to pull this section of the readme and add it to the page?

hm yeah, technically. I think this is better pulled from the `package.json` `description` field.

**Review by xelaint — APPROVED** [review]

**Pull request merged**